scene/gui/line_edit.cpp

```diff
@@ -372,6 +372,11 @@ void LineEdit::gui_input(const Ref<InputEvent> &p_event) {
 				selection.drag_attempt = false;
 			}
 
+			if (select_all_on_focus && pending_select_all_on_focus) {
```

Comment:
I think checking `select_all_on_focus` here is not required, because `pending_select_all_on_focus` already requires it. If `select_all_on_focus` gets changed between the click and release, you are left with lingering `pending_select_all_on_focus`.